tests/Go/PhpUnit/ClassMemberNotWovenConstraint.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/*
+ * Go! AOP framework
+ *
+ * @copyright Copyright 2011, Lisachenko Alexander <lisachenko.it@gmail.com>
+ *
+ * This source file is subject to the license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Go\PhpUnit;
+
+use Go\TestUtils\AdvisorIdentifiersExtractor;
+use PHPUnit_Framework_Constraint as Constraint;
+use ReflectionClass;
+use Go\Instrument\PathResolver;
+
+/**
+ *Asserts that class member is not woven for given class.
+ */
+class ClassMemberNotWovenConstraint extends Constraint
+{
+    /**
+     * @var array
+     */
+    private $configuration;
+
+    public function __construct(array $configuration)
+    {
+        parent::__construct();
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function matches($other)
+    {
+        if (!$other instanceof ClassAdvisorIdentifier) {
+            throw new \InvalidArgumentException(sprintf('Expected instance of "%s", got "%s".', ClassAdvisorIdentifier::class, is_object($other) ? get_class($other) : gettype($other)));
+        }
+
+        $wovenAdvisorIdentifiers = $this->getWovenAdvisorIdentifiers($other->getClass());
+
+        $target = $other->getTarget();
+
+        if (!isset($wovenAdvisorIdentifiers[$target])) {
+            return true;
+        }
+
+        if (!isset($wovenAdvisorIdentifiers[$target][$other->getSubject()])) {
+            return true;
+        }
+
+        if (null === $other->getAdvisorIdentifier()) {
+            return false; // if advisor identifier is not specified, that means that any matches, so weaving exists.
+        }
+
+        return !in_array($other->getAdvisorIdentifier(), $wovenAdvisorIdentifiers[$target][$other->getSubject()], true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toString()
+    {
+        return 'join point does not exists.';
+    }
+
+    /**
+     * Get woven advisor identifiers.
+     *
+     * @param string $class
+     * @return array
+     */
+    private function getWovenAdvisorIdentifiers($class)
+    {
+        ClassLocator::initialize($this->configuration);
+
+        $advisorIdentifiers = (new ReflectionClass($class))->getStaticPropertyValue('__joinPoints')->getValue();
```

Comment:
Ok, now I am 100% sure, `goaop\parser-reflection` can not be used for this task. Its usage is fragile, if for any reason original class is loaded into memory, every successive test will fail. I have jumped over hoops to make it work - it is useless, sorry, I am not able deliver that in such way.  

In regards to using `ReflectionEngine` methods for parsing classes, I do not have any benefit from it, I will get a raw AST which I will have to eventually traverse by my self manually. Tree walker gives my much more control and I can solve problem easier with less code.

In regards of using eval, since you are against it, I will try to use `NodeExpressionResolver` instead. If it is possible to make it work, fine, if not, eval it is, sorry.